framework/src/headers/ip/v6/srh.rs

```diff
@@ -99,64 +101,87 @@ use utils::*;
       path and so on.
 
    o  Type Length Value (TLV) are described in Section 3.1.
-*/
+ */
+
+// As per Spec: Routing Type: TBD, to be assigned by IANA (suggested value: 4).
+pub const ROUTING_TYPE: u8 = 4;
 
 // Positions of the various flag bits in the flag byte
 const PROTECTED_FLAG_POS: u8 = 1;
 const OAM_FLAG_POS: u8 = 2;
 const ALERT_FLAG_POS: u8 = 3;
 const HMAC_FLAG_POS: u8 = 4;
 
-// SRv6 Segment IDs are an array of 128-bit values with a length defined at
-// runtime.
-pub type Segments = [u128];
+pub type Segment = Ipv6Addr;
+pub type Segments = [Segment];
+
+// Generic Type for parsing/reading possible variable length segment list.
+pub type U_ = U0;
 
 // The v6 Segment Routing Header is a specialization of the v6 Routing Header,
 // defined in
 // https://tools.ietf.org/html/draft-ietf-6man-segment-routing-header-11. Like
 // all extension headers, it shares the first two fields.
-#[derive(Default)]
+#[derive(Debug)]
 #[repr(C, packed)]
-pub struct SegmentRoutingHeader<T>
+pub struct SegmentRoutingHeader<T, S>
 where
     T: Ipv6VarHeader,
+    S: ArrayLength<Segment>,
 {
-    ext_header: Ipv6ExtHeader<T>,
+    pub ext_header: Ipv6ExtHeader<T>,
     routing_type: u8,
     segments_left: u8,
     last_entry: u8,
     flags: u8,
     tag: u16, // Segments and TLVs follow this, but must be accessed via raw pointers
+    segments: GenericArray<Segment, S>,
 }
 
 // SRv6 can encapsulate any L4 IP protocol.
-impl<T> IpHeader for SegmentRoutingHeader<T>
+impl<T, S> IpHeader for SegmentRoutingHeader<T, S>
 where
     T: Ipv6VarHeader,
+    S: ArrayLength<Segment>,
 {
 }
 
 // The SegmentRoutingHeader is an extension header, and so has a next header
 // field and can be the PreviousHeader for another header.
-impl<T> Ipv6VarHeader for SegmentRoutingHeader<T>
+impl<T, S> Ipv6VarHeader for SegmentRoutingHeader<T, S>
 where
     T: Ipv6VarHeader,
+    S: ArrayLength<Segment>,
 {
     fn next_header(&self) -> Option<NextHeader> {
         self.ext_header.next_header()
     }
 }
 
 // Formats the header for printing
-impl<T> fmt::Display for SegmentRoutingHeader<T>
+impl<T, S> fmt::Display for SegmentRoutingHeader<T, S>
 where
     T: Ipv6VarHeader,
+    S: ArrayLength<Segment>,
 {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut segs: String = String::new();
+        let segments = self.segments();
+        let num_segments = segments.len();
+
+        for (idx, s) in segments.iter().enumerate() {
+            let addy = s.to_string();
+            segs.push_str(&addy);
+
+            if idx != num_segments - 1 {
+                segs.push_str(", ")
```

Comment:
Isn't there a `.join(",")` in rust?

https://doc.rust-lang.org/std/slice/trait.SliceConcatExt.html#tymethod.connect